service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ReviewCMOControllerSubmittedTest.java

```diff
@@ -95,13 +102,19 @@ void shouldSendCMOIssuedNotificationsIfJudgeApproves() {
 
         CaseDetails caseDetails = buildCaseDetailsForApprovedCMO();
         caseDetails.setId(CASE_ID);
+        caseDetails.setJurisdiction(JURISDICTION);
+        caseDetails.setCaseTypeId(CASE_TYPE);
 
         CaseDetails caseDetailsBefore = CaseDetails.builder().data(
             Map.of("draftUploadedCMOs", List.of(element(buildCMO(SEND_TO_JUDGE))))).build();
 
         CallbackRequest callbackRequest = CallbackRequest.builder().caseDetails(caseDetails)
             .caseDetailsBefore(caseDetailsBefore).build();
 
+        CaseData caseData = mapper.convertValue(callbackRequest.getCaseDetails().getData(), CaseData.class);
```

Comment:
what if you change provate method to 

` private CaseDetails buildCaseDetailsForApprovedCMO(CaseManagementOrder... caseManagementOrders) {
        UUID cmoId = UUID.randomUUID();

        CaseDetails caseDetails = asCaseDetails(CaseData.builder()
            .representatives(createRepresentatives())
            .caseLocalAuthority(LOCAL_AUTHORITY_CODE)
            .draftUploadedCMOs(List.of(element(cmoId, buildCMO(SEND_TO_JUDGE))))
            .sealedCMOs(ElementUtils.wrapElements(caseManagementOrders))
            .reviewCMODecision(buildReviewDecision(SEND_TO_ALL_PARTIES))
            .hearingDetails(List.of(element(hearing(cmoId))))
            .build());
        caseDetails.setId(CASE_ID);
        caseDetails.setJurisdiction(JURISDICTION);
        caseDetails.setCaseTypeId(CASE_TYPE);
        return caseDetails;
    }`

and then test can be like:
` given(documentDownloadService.downloadDocument(order.getBinaryUrl())).willReturn(DOCUMENT_CONTENT);

        CaseManagementOrder caseManagementOrder = buildCMO(APPROVED);

        CaseDetails caseDetails = buildCaseDetailsForApprovedCMO(caseManagementOrder);

        CaseDetails caseDetailsBefore = CaseDetails.builder().data(
            Map.of("draftUploadedCMOs", List.of(element(buildCMO(SEND_TO_JUDGE))))).build();

        CallbackRequest callbackRequest = CallbackRequest.builder().caseDetails(caseDetails)
            .caseDetailsBefore(caseDetailsBefore).build();


        postSubmittedEvent(callbackRequest);

        checkUntil(() -> {
       ....

            verify(coreCaseDataService).triggerEvent(JURISDICTION,
                CASE_TYPE,
                CASE_ID,
                SEND_DOCUMENT_EVENT,
                Map.of("documentToBeSent", caseManagementOrder.getOrder()));

            verifyNoMoreInteractions(notificationClient);
`

Reply:
nice suggestion will change :)